torch/nn/_functions/thnn/upsampling.py

```diff
@@ -62,15 +59,39 @@ def backward(self, grad_output):
         return grad_input
 
 
-class UpsamplingBilinear2d(_UpsamplingBase):
+class UpsamplingBilinear2d(Function):
+
+    def __init__(self, size=None, scale_factor=None):
+        super(UpsamplingBilinear2d, self).__init__()
+        if size is None and scale_factor is None:
+            raise ValueError('either size or scale_factor should be defined')
+        if scale_factor is not None:
+            if not isinstance(scale_factor, (Integral, tuple)):
+                raise ValueError('scale_factor must be a non-negative integer, or a tuple of non-negative integers')
+            if isinstance(scale_factor, tuple):
+                try:
+                    assert len(scale_factor) == 2
+                    for i in scale_factor:
+                        assert isinstance(i, Integral)
+                        assert i >= 1
+                except AssertionError as e:
+                    raise ValueError('scale_factor must be a non-negative integer, or a tuple of non-negative integers')
+                
+        if size is not None and not isinstance(size, tuple):
+            size = (size, size)
+        if scale_factor is not None and not isinstance(scale_factor, tuple):
+            scale_factor = (scale_factor, scale_factor)
```

Comment:
Thanks for pointing this out!  I used it elsewhere but can do it here too.

I ran into some problems when the tests passed `None` for `scale_factor`; the code would pass in (None, None) and then the integer check would fail.  I'll take another pass at this in particular.